tests/test_warn_on_error.py

```diff
@@ -10,7 +11,12 @@ def test_check_type(recwarn):
         check_type(1, str, typecheck_fail_callback=warn_on_error)
 
     assert len(warning.list) == 1
-    assert warning.list[0].filename == __file__
+    if (
+        os.name == "nt"
+    ):  # file path name (and drive letter) is case insensitive on windows
+        assert warning.list[0].filename.lower() == __file__.lower()
+    else:
+        assert warning.list[0].filename == __file__
```

Comment:
What exactly prompted this change? Was something failing?

Reply:
It break test on Windows. The drive letter (`D:\`) can be either lower or upper case. And they do appear differently two sides sometime.

However, they do not always appear differently. So this sometimes make the tests fail on my windows machine. I do know how to definitely reproduce this. Windows is case insensitive. Maybe `os.path.normcase` can be used avoid the test for OS.